src/stores/modules/Groups.store.js

```diff
@@ -0,0 +1,93 @@
+import api from '../../api/index';
+
+const state = {
+  groups: [],
+  valid: false,
+  fetching: true
+};
+
+// getters https://vuex.vuejs.org/guide/getters.html
+
+const getters = {};
+
+// actions https://vuex.vuejs.org/guide/actions.html
+//Actions are similar to mutations, the differences being that:
+//
+//	Instead of mutating the state, actions commit mutations.
+//	Actions can contain arbitrary asynchronous operations.
+
+const actions = {
+  GET_GROUPS ({ commit, rootState }, groupName) {
+    commit('fetching');
+    return new Promise((resolve, reject) => {
+      api.groups.getGroups(groupName, rootState.User.JWT)
+        .then(response => response.json())
+        .then(json => {
+          if (!json.success) {
+            return reject(json.message || json.messages);
+          }
+          commit('receiveGroups', json);
+          resolve(json);
+        });
```

Comment:
is a `.catch` required too?

(Same for the other actions below)

Reply:
Yes, that's part of story I've added to kanban .. error handling from api through to the view.

Reply:
Excellent.